src/chainstate/stacks/db/mod.rs

```diff
@@ -753,11 +789,15 @@ impl StacksChainState {
             }
 
             if db_config.version != CHAINSTATE_VERSION {
-                error!(
-                    "Invalid chain state database: expected version = {}, got {}",
-                    CHAINSTATE_VERSION, db_config.version
-                );
-                return Err(Error::InvalidChainstateDB);
+                if db_config.version == "1" {
```

Comment:
I almost wonder if this block should be its own method (or possibly its own module) that just executes migrations.  We'll surely add/change more tables in the future.

Related, do we have a plan or list of recommendations for schema changes that alter existing tables?  While we could always use `ALTER TABLE` to add new columns, I'm not sure that would result in the most maintainable code.  The way I've seen this done in other systems is they'll let developers create whatever schemas they want in-between successive versions, and then have their own `migrations` module that automatically calculates the schema delta between two schemas and executes code to morph one schema into another.  We don't have to solve this right now, but it's something to think about (I'll need to figure out something for #1805 because it adds new tables as well as new fields to existing tables).

Also, I think there should be a unit test for this (and for all schema migrations), so we can ensure that a node started from any DB schema version can automatically migrate to whatever the node's current schema version is.

Reply:
> I almost wonder if this block should be its own method (or possibly its own module) that just executes migrations. We'll surely add/change more tables in the future.

Probably could be, though my preference is always to save this kind of abstraction for when we have something to abstract across (i.e., more than one example). Otherwise, I worry about abstracting wrongly.

>  The way I've seen this done in other systems is they'll let developers create whatever schemas they want in-between successive versions, and then have their own migrations module that automatically calculates the schema delta between two schemas and executes code to morph one schema into another. We don't have to solve this right now, but it's something to think about (I'll need to figure out something for #1805 because it adds new tables as well as new fields to existing tables).

Yeah -- I've thought a little bit about this, and I think we ultimately will need something pretty generic. Looking at the last few db schema changes:

1. the addition of `parent_sortition_id` to the sortition db
2. new indexes
3. db_schema version table
4. this PR

Items 2, 3, and 4, would all be potentially solvable with a schema-computed migration. Item 1 _wouldn't_ be though, and neither would many of the changes for #1805. For those, migration may be possible with a bulk process (i.e., `parent_sortition_id` could have been written from available state), or it may not be, in which case, the migration process should just panic.

I think because of that we likely will want a pretty abstracted process, where schema versions just have associated functions that handle the migration from `vPrior -> vCurrent`, and we iterate those. If it turns out that many of those handlers could be rewritten with schema-computed migrations, then we can refactor those handlers, but I don't think we should try to put something like that in place before it's clear that there would be a benefit.

Reply:
Sounds good to me!